src/runtime/composables/useSupabaseUser.ts

```diff
@@ -1,22 +1,6 @@
 import type { User } from '@supabase/supabase-js'
-import { useSupabaseClient } from './useSupabaseClient'
 import { useState } from '#imports'
 
 export const useSupabaseUser = () => {
-  const supabase = useSupabaseClient()
-
-  const user = useState<User | null>('supabase_user', () => null)
-
-  // Asyncronous refresh session and ensure user is still logged in
-  supabase?.auth.getSession().then(({ data: { session } }) => {
```

Comment:
"useSupabaseUser() relies on useState(), which is updated on auth events from the supabase client" => Why not keeping this async call to also relies on the useSupabaseUser call ?